<?xml version="1.0"?>
<launch>
    <arg name="debug" default="false" /> 
    <arg unless="$(arg debug)" name="launch_prefix" value="" /> 
    <arg if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />
    <!-- GDB Debug Option -->

    <node pkg="kortex_control" type="master_controller" name="master_controller" output="screen"  launch-prefix="$(arg launch_prefix)">
        <param name="robot_description" command="xacro $(find kortex_control)/arm_description/robot_model.xacro" />
        <param name="base_frame" value="base_link" />
        <param name="tip_frame" value="tool_frame" />
        <param name="ee_marker_frame" value="ee_marker_frame" />
        <param name="target_frame" value="dummy_pelvis" />
    </node>

    <node pkg="kortex_control" type="tf_listener" name="pelvis_tf_listener" output="screen" >
        <param name="source_frame" value="tool_frame" />
        <param name="target_frame" value="dummy_pelvis" />
    </node>

    <node pkg="kortex_control" type="tf_listener" name="camera_tf_listener" output="screen" >
        <param name="source_frame" value="ee_marker_frame" />
        <param name="target_frame" value="dummy_camera" />
    </node>

    <node pkg="kortex_gazebo" type="dummy_pelvis_tf" name="dummy_pelvis_tf" output="screen" />
    <node pkg="kortex_gazebo" type="dummy_camera_tf" name="dummy_camera_tf" output="screen" />

</launch>
